R/getRunStatus.R

```diff
@@ -90,106 +93,97 @@ getRunStatus <- function(mydir = dir(), sort = "nf", user = NULL) {
-        if (exists("stats")) if (any(grepl("config", names(stats)))) if (stats[["config"]][["model_name"]] == "MAgPIE") {
-          ovdir <- "/p/projects/rd3mod/models/results/magpie/"
-        } else {
-          ovdir <- "/p/projects/rd3mod/models/results/remind/"
+    if (onCluster) {
+      out[i, "runInAppResults"] <- "no"
+      if (file.exists(fle)) {
+        load(fle)
+        if (any(grepl("id", names(stats)))) {
+          if (exists("stats") && any(grepl("config", names(stats))) && stats[["config"]][["model_name"]] == "MAgPIE") {
+            ovdir <- "/p/projects/rd3mod/models/results/magpie/"
+          } else {
+            ovdir <- "/p/projects/rd3mod/models/results/remind/"
+          }
+          try(id <- paste0(ovdir, stats[["id"]], ".rds"))
+          if (file.exists(id) && all((file.info(Sys.glob(paste0(ovdir, "overview.rds")))$mtime + 600) > file.info(id)$mtime)) {
+            out[i, "runInAppResults"] <- "yes"
+          }
         }
-        try(id <- paste0(ovdir, stats[["id"]], ".rds"))
-        if (exists("ovdir")) if (file.exists(id) && all((file.info(Sys.glob(paste0(ovdir, "overview.rds")))$mtime + 600) > file.info(id)$mtime))
-          out[i, "runInAppResults"] <- TRUE
-      } else {
-        if (onCluster) out[i, "runInAppResults"] <- FALSE
       }
     }
 
     # Iter
-    if (exists("cfg")) totNoOfIter <- cfg[["gms"]][["cm_iteration_max"]]
+    cm_iteration_max <- cfg$gms$cm_iteration_max
+    if (isTRUE(cfg$gms$cm_nash_autoconverge > 0) && grepl("nash", out[i, "RunType"])) {
+      if (file.exists(fullgms)) {
+        cm_iteration_max <- suppressWarnings(system(paste0("tac ", fullgms, "| grep -m 1 'cm_iteration_max = [1-9].*.;[ ]*$'"), intern = TRUE))
+        cm_iteration_max <- sub(";[ ]*", "", sub("^.*.= ", "", cm_iteration_max))
+      }
+    }
     out[i, "Iter"] <- "NA"
     out[i, "RunStatus"] <- "NA"
     if (file.exists(fulllog)) {
       suppressWarnings(try(loop <- sub("^.*.= ", "", system(paste0("grep 'LOOPS' ", fulllog, " | tail -1"), intern = TRUE)), silent = TRUE))
       if (length(loop) > 0) out[i, "Iter"] <- loop
-      if (!out[i, "RunType"] %in% c("nash", "Calib_nash") & length(totNoOfIter) > 0) out[i, "Iter"] <- paste0(out[i, "Iter"], "/", sub(";", "", sub("^.*.= ", "", totNoOfIter)))
+      if (length(cm_iteration_max) > 0) out[i, "Iter"] <- paste0(out[i, "Iter"], "/", cm_iteration_max)
       suppressWarnings(try(out[i, "RunStatus"] <- substr(sub("\\(s\\)", "", sub("\\*\\*\\* Status: ", "", system(paste0("grep '*** Status: ' ", fulllog), intern = TRUE))), start = 1, stop = 17), silent = TRUE))
       if (onCluster & out[i, "RunStatus"] == "NA") {
-        if (out[i, "jobInSLURM"] == FALSE) {
+        if (out[i, "jobInSLURM"] == "no") {
           out[i, "RunStatus"] <- "Run interrupted"
         } else {
           out[i, "RunStatus"] <- "Run in progress"
-          }
+        }
       } else {
         if (out[i, "RunStatus"] == "NA") out[i, "RunStatus"] <- "Run interrupted"
```

Comment:
- for coupled runs, find out whether MAgPIE is still running and print that to the user instead of `Normal Completion` because REMIND is already done.